app/controllers/concerns/enlistment_filters.rb

```diff
@@ -76,8 +76,33 @@ def build_code_location
   def project_has_code_location?
     return unless CodeLocationSubscription.code_location_exists?(@project.id, @code_location.url,
                                                                  @code_location.branch, code_location_params[:scm_type])
-
-    flash[:notice] = t('.notice', url: @code_location.url, module_branch_name: @code_location.branch)
+    manage_deleted_enlistment
     redirect_to project_enlistments_path(@project)
   end
+
+  def manage_deleted_enlistment
+    if enlistment_deleted?
+      restore_deleted_enlistment
+    else
+      flash[:notice] = t('.notice', url: @code_location.url, module_branch_name: @code_location.branch)
+    end
+  end
+
+  def enlistment_deleted?
+    enlistment = get_enlistment
+    enlistment.try(:deleted?) ? true : false
+  end
+
+  def get_code_location_id
+    CodeLocationApi.id_from_url(@code_location.url, @code_location.branch)
```

Comment:
This needs to be updated to use the CodeLocation GET call with the url and branch and then extract the ID from the return package.